core/Pos/Binary/Core/Address.hs

```diff
@@ -138,9 +138,14 @@ instance Bi (Attributes AddrAttributes) where
             , aaStakeDistribution = BootstrapEraDistr
             }
         go n v acc =
+            -- [CSL-1493] Should we:
+            -- 1. fail explicitly here (by using risky unsafe functions)
+            -- 2. Yield `Nothing` in case we cannot deserialize `v`
+            -- 3. Make `go` and thus `decodeAttributes` take an impure `updater`
+            --    function (which runs in the `Decoder` monad).
```

Comment:
(1) is bad for reasons I described in the issue.
(2) is bad because I think we should distinguish cases when we don't want to decode something and we have malformed data.
(3) seems to be good, I don't see disadvantages.